### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by naris-silpakit**
*Issue #, if available:*
N/A

*Description of changes:*
Adds content to the `teams.md` documentation file, fixes broken links in the documentation, and updates the table under `docs/add-ons/index.md`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by vara-bonthu — APPROVED** [review]
Well done on the first PR Naris!  
Docs looks good to me 👍🏼

File: docs/teams.md [hunk 1]
```diff
@@ -1 +1,125 @@
...
+- Registers an existing role (or create a new role) for cluster access with trust relationship with the provided/created role
+
+## Team Benefits
+
+By managing teams via infrastructure as code, we achieve the following benefits:
+
+1. Self-documenting code
+2. Centralized logic related to the team
+3. Clear place where to add additional provisioning, for example adding Kubernetes Service Accounts and/or infrastructure, such as S3 buckets
+4. IDE support to locate the required team, e.g. CTRL+T in VSCode to lookup class name.
```
Comment (Zvikan) [inline docs/teams.md:113]: Can be removed, this is true for the CDK

File: docs/teams.md [hunk 2]
```diff
@@ -1 +1,125 @@
...
+  }
+```
+
+The `application_teams` will do the following for every provided team:
+
+- Create a namespace
+- Register quotas
+- Register IAM users for cross-account access
+- Create a shared role for cluster access. Alternatively, an existing role can be supplied.
+- Register provided users/role in the `awsAuth` map for `kubectl` and console access to the cluster and namespace.
```
Comment (Zvikan) [inline docs/teams.md:81]: "`aws-auth` configmap for..."   -  should probably fix it on the module README as-well

**Review by Zvikan — COMMENTED** [review]
2 little comments but other than that LGTM ! thank you !

File: docs/teams.md (hunk 1 above)
Comment (naris-silpakit) [reply docs/teams.md:113]: Whoops - removed.

File: docs/teams.md (hunk 2 above)
Comment (naris-silpakit) [reply docs/teams.md:81]: Fixed in both files.

**Pull request merged**